packages/nimble-components/src/select/index.ts

```diff
@@ -622,9 +645,36 @@ export class Select
         // implementation handles skipping non-selected disabled options for the initial
         // selected value.
         this.setSelectedOptions();
+        if (this.open) {
+            this.setActiveOption(this.selectedIndex);
+        }
         this.updateValue();
     }
 
+    /**
+     * @internal
+     * Fork of Listbox implementation, so that the selectedIndex is not changed while the dropdown
+     * is open.
+     */
+    public override typeaheadBufferChanged(_: string, __: string): void {
+        if (this.$fastController.isConnected) {
+            const typeaheadMatches = this.getTypeaheadMatches();
+
+            if (typeaheadMatches.length) {
+                const selectedIndex = this.options.indexOf(
```

Comment:
Similar to one of my other comments, can this be called something other than `selectedIndex`? Only in some code paths will this item actually become selected. Perhaps something more like `matchIndex` would be more appropriate.